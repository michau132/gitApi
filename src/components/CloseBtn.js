import React from 'react';
import styled from 'styled-components';
import { Link } from 'react-router-dom';

const CloseBtnStyles = styled.div`
  position: fixed;
  right: 40px;
  top: 25px;
  width: 32px;
  height: 32px;
  z-index: 5;
  cursor: pointer;
  background-color: #333;
  border-radius: 4px;
  
  :hover {
    opacity: 0.5;
  }
  :before, :after {
    content: ' ';
    position: absolute;
    left: 15px;
    height: 33px;
    width: 2px;
    background-color: #fff;
    border-radius: 15px;
  }
  :before {
    transform: rotate(45deg);
  }
  :after {
    transform: rotate(-45deg);
  }
  
  ${({ theme }) => theme.media.desktop} {
    position: absolute;
  }
`;

const CloseBtn = () => (
  <Link to={{ pathname: '/trainingGit/', state: { modal: false } }}>
    <CloseBtnStyles />
  </Link>

);

export default CloseBtn;
