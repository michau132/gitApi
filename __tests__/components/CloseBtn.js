import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import CloseBtn from '../../src/components/CloseBtn';

describe('<CloseBtn /> component', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<CloseBtn />);
  });

  test('renders without crashing', () => {
    expect(wrapper).toBeDefined();
  });

  test('matching snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
