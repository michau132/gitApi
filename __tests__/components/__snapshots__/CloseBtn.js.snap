// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CloseBtn /> component matching snapshot 1`] = `
<Link
  replace={false}
  to={
    Object {
      "pathname": "/",
      "state": Object {
        "modal": false,
      },
    }
  }
>
  <styled.div />
</Link>
`;
